### PR TITLE
Testing suite for gaussian.py and samba_models by extension

### PR DESCRIPTION
**Pull request by asemposki**
Here is the testing suite I wrote with five test functions, each of which tests the evaluation and predictions of my BMM code when given a specific SAMBA model with a truncation at third order in both models. This may need to be altered later, but for now it can compare to the data in samba_results.txt to check itself. 

I have one warning that I have not figured out yet, but other than that all tests are passing.

### CODE REVIEW

File: test/test_gaussian.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+posterior_draws, mixed_mean, mixed_intervals, std_dev = mixed.predict(ci=ci)
+
+# now call for evaluation of the weights
+weights = mixed.evaluate_weights()
+
+###########################################################
+# testing suite
+###########################################################
+
+def test_models():
```
[the remark this answers is not in the document]
Comment (danOSU) [reply test/test_gaussian.py:56]: @asemposki can you see this?

**Review by danOSU — CHANGES_REQUESTED** [review]

File: test/test_gaussian.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        "incorrect evaluation for large-g"
+
+def test_init():
+
+    # check passing of variables into Multivariate class
+    assert mixed.model_dict == models, "class variable self.model_dict not set"
+    assert mixed.n_models == n_models, "class variable self.n_models not set"
+    assert np.array_equal(mixed.x, g), "class variable self.x not set"
+
+def test_mixing():
```
[the remark this answers is not in the document]
Comment (asemposki) [reply test/test_gaussian.py:98]: test_mixing() tests that the mixed model is yielding something back that is not None, but it is not testing that the results of the mixed model are equal to that of SAMBA's mixing. I can add to the samba_results.txt file the mixed model and use that here.

File: test/test_gaussian.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    # check returned variables
+    assert posterior_draws == 0.0, "posterior draws is not zero"
+    assert mixed_mean is not None, "mixed_mean is None"
+    assert mixed_intervals is not None, "mixed_intervals is None"
+    assert std_dev is not None, "std_dev is None"
+
+def test_evaluate_weights():
+
+    # now that predict has been run, test the weights
+    assert np.array_equal(weights, mixed.var_weights), "weights are incorrect"
```
[the remark this answers is not in the document]
Comment (asemposki) [reply test/test_gaussian.py:114]: No, the weights come from the mixing. They are my precision values from the way that I perform multivariate mixing, so they do not come from the input models alone. Hence they are part of the mixing class for me and they are generated while mixing, so the evaluate_weights() function returns their values from when mixing has been performed.

**Comment by asemposki** [comment]
Reopening the PR.

**Pull request merged**